### PR TITLE
handled queries to endpoint /api/reviews

### PR DESCRIPTION
**Pull request by MrDonev**
11- reviews queries

### CODE REVIEW

**Comment by sirKevlar** [comment]
I would ideally like to see 2 more sad path tests here... One for invalid sort_by and one for invalid order. Also we need a test for a review which has no comments returning an empty array

File: models/reviews.model.js [hunk 1]
```diff
@@ -45,29 +44,73 @@ exports.updateReviewById = (reviewId, updateVotes) => {
...
+  }
+  const tableQuery = db.query(queryStr, queryVals);
+  const allCategories = db.query(`SELECT * FROM categories`);
+  return Promise.all([tableQuery, allCategories]).then(
+    ([resultTable, resultCategories]) => {
+      const endpointQuery = resultTable.rows;
+      const categoryCheck = resultCategories.rows.map((obj) => obj.slug);
+      if (endpointQuery.length) {
+        return endpointQuery.map(review=>{
+          review.comment_count=Number(review.comment_count)
```
Comment (emilyb93) [inline models/reviews.model.js]: I would prefer you'd be changing the data type of comment_count in the SQL query with either CAST() or ::int, have a look at these and see if you can refactor to remove this step. This is quite inefficient because you now have to loop over every single one of your results which could be hundreds of reviews long

File: models/reviews.model.js [hunk 2]
```diff
@@ -45,29 +44,73 @@ exports.updateReviewById = (reviewId, updateVotes) => {
...
+    GROUP BY reviews.review_id
+    ORDER BY ${sort_by} ${order_by}`;
+    queryVals.push(category);
+  }
+  const tableQuery = db.query(queryStr, queryVals);
+  const allCategories = db.query(`SELECT * FROM categories`);
+  return Promise.all([tableQuery, allCategories]).then(
+    ([resultTable, resultCategories]) => {
+      const endpointQuery = resultTable.rows;
+      const categoryCheck = resultCategories.rows.map((obj) => obj.slug);
```
Comment (emilyb93) [inline models/reviews.model.js]: I would prefer you did this check earlier, as by this point, you could have already been subject to SQL injection.

File: models/reviews.model.js [hunk 3]
```diff
@@ -92,8 +135,8 @@ exports.fetchReviewCommentsById = (id) => {
 };
 
 exports.addCommentByReviewId = (reviewId, newComment) => {
-  if (Object.keys(newComment).length < 2){
-    return Promise.reject({status:400, msg: 'Bad Request'})
+  if (Object.keys(newComment).length < 2) {
```
Comment (emilyb93) [inline models/reviews.model.js]: Id prefer you check for the exact keys you are after here. This could be 2 completely different and incorrect keys and still pass

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Some changes to think about

File: models/reviews.model.js (hunk 2 above)
Comment (MrDonev) [reply models/reviews.model.js]: I don't quite get which check should I refactor. Could you please specify?

File: models/reviews.model.js [hunk 4]
```diff
@@ -95,22 +95,22 @@ exports.fetchAllReviews = (
...
-        return [];
       }
-    }
-  );
+    })
+    .then(() => {
+      return tableQuery
+        .then(({ rows }) => {
+          return rows;
+        })
```
Comment (emilyb93) [inline models/reviews.model.js]: This is the right order but now we are nesting our .thens, try to keep it all in one promise chain at the same level

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Change nesting in promise chain

**Review by emilyb93 — APPROVED** [review]
Brill
